tests/framework/installer/ceph_installer.go

```diff
@@ -56,15 +58,42 @@ type CephInstaller struct {
 	T                func() *testing.T
 }
 
-func (h *CephInstaller) CreateCephCRDs() (err error) {
+func (h *CephInstaller) CreateCephCRDs() error {
 	var resources string
 	logger.Info("Creating Rook CRDs")
 
 	resources = h.Manifests.GetRookCRDs()
 
-	_, err = h.k8shelper.KubectlWithStdin(resources, createFromStdinArgs...)
+	var err error
+	for i := 0; i < 5; i++ {
+		if i > 0 {
+			logger.Infof("waiting 10s...")
+			time.Sleep(10 * time.Second)
+		}
+
+		_, err = h.k8shelper.KubectlWithStdin(resources, createFromStdinArgs...)
+		if err == nil {
+			return nil
+		}
+
+		// If the CRD already exists, the previous test must not have completed cleanup yet.
+		// Delete the CRDs and attempt to wait for the cleanup.
+		if strings.Index(err.Error(), "AlreadyExists") == -1 {
+			return err
+		}
 
-	return
+		// remove the finalizer from the cluster CRD
+		if _, err := h.k8shelper.Kubectl("patch", "crd", "clusters.ceph.rook.io", "-p", `{"metadata":{"finalizers": []}}`, "--type=merge"); err != nil {
+			logger.Warningf("could not remove finalizer from cluster crd. %+v", err)
+		}
+
+		logger.Warningf("CRDs were not cleaned up from a previous test. Deleting them to try again...")
+		if _, err := h.k8shelper.KubectlWithStdin(resources, deleteFromStdinArgs...); err != nil {
```

Comment:
we don't already have a function that cleans up the ceph CRDs? should we unify the way that we clean them up during teardown and the way we are cleaning them up in this retry loop?

Reply:
I was hesitant to remove the finalizer as a normal part of test teardown because the finalizer might actually be blocking on something that should be performed. This way we only remove the finalizer if it is blocking a new test from starting.